### DISCUSSION TITLE
Get remotes buttonevent

### DISCUSSION THREAD

**Discussion by mariusmotea** [Koenkk/zigbee2mqtt#7498] [General]
Hi,

In last months i work on integrating ZigBee2Mqtt with DiyHue in order to have a similar experience with the original Hue Bridge. One issue that i have is that i need conversion for every button of every device. ZigBee2Mqtt provide more human frienfly events, example: "on-press, on-hold, on-hold-release, up-press, up-hold, up-hold-release" for the Hue dimmer switch.  On Hue API these events are some integer codes, exemple : `{"buttonevent": 1002}`. I believe Hue Bridge provide RAW event code from the remotes and this is very good because based on this integer code rules are processed on the bridge to trigger related events.
In case it is true what i said, is there a way to receive RAW button code form remotes?

Regards.

**Comment by Koenkk** [comment]
Receiving raw button codes is not supported (and makes no sense IMHO). Note that z2m sends all event, just with a friendly name instead of the raw code.

**Reply by mariusmotea** [reply]
Well, most Hue apps crash if they get a string instead of the expected integer value on the `buttonevent` key so i cannot store the friendly name. Now i revert friendly names to correct integer value using an array, but i can do this only for devices i know, not for all. HueEssentials app is able to configure much more devices because it also work with Deconz, but the applied rules are based on the correct buttonevent.
